examples/paho-mqtt/Makefile

```diff
@@ -33,25 +33,20 @@ USEMODULE += ps
 USEMODULE += netdev_default
 USEPKG += paho-mqtt
 
-# paho-mqtt depends on TCP support, choose which stacks you want
-LWIP_IPV4 ?= 0
-LWIP_IPV6 ?= 1
-
-ifneq (0,$(LWIP_IPV4))
-  USEMODULE += ipv4_addr
-  USEMODULE += lwip_arp
-  USEMODULE += lwip_ipv4
-  USEMODULE += lwip_dhcp_auto
-  CFLAGS += -DETHARP_SUPPORT_STATIC_ENTRIES=1
-endif
-
-ifneq (0,$(LWIP_IPV6))
-  USEMODULE += ipv6_addr
-  USEMODULE += lwip_ipv6_autoconfig
-endif
-
-USEMODULE += lwip_netdev
-USEMODULE += lwip
+USEMODULE += auto_init_gnrc_netif
+# Activate ICMPv6 error messages
+USEMODULE += gnrc_icmpv6_error
+# Specify the mandatory networking module for a IPv6 routing node
+USEMODULE += gnrc_ipv6_router_default
+# Add a routing protocol
+USEMODULE += gnrc_rpl
+USEMODULE += auto_init_gnrc_rpl
+# Additional networking modules that can be dropped if not needed
+USEMODULE += gnrc_icmpv6_echo
```

Comment:
Why not support both?
e.g. `examples/gcoap` and [`examples/telnet_server`](https://github.com/RIOT-OS/RIOT/blob/master/examples/telnet_server/Makefile#L14) can be compiled with either GNRC or LWIP.

Reply:
Sure it would, but what would be the advantage? In my understanding the example applications should be easy to use (and as simple as possible). I think having one example that demonstrates how to build a network application for different stacks is sufficient.